real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/PlayBackUtil.java

```diff
@@ -30,5 +65,110 @@ public static boolean validateLogFileMetadata(JsonObject logFile) {
 
         return true;
     }
+    
+
+    /**
+     * reads a sensor log and outputs the next message, if the next message is not suppose to appear for some time then this method will block until that time
+     * @param sensorDataJson the json object of sensor data
+     * @param translator translator object
+     * @param playBacktime the time the playback should play until
+     * @param sensorStartTime the time the sensor playback started at
+     * @param playBackSpeed the speed to playback at
+     * @return the message from the log
+     * @throws InterruptedException timing didn't work
+     */
+    public static Message parseSensorLog(JsonObject sensorDataJson,Gson translator,long playBacktime,
+    		long sensorStartTime,double playBackSpeed) throws InterruptedException{
+            // wait until the time this message is supposed to be sent
+			long sensorTime = sensorDataJson.get("timestamp").getAsLong();
+            long sensorDt = (sensorTime-sensorStartTime);
+            long dt = (long) (sensorDt/playBackSpeed) - playBacktime;
+            if(dt> 10){ //Milliseconds
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using Literals in Conditional Statements](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/2662564685/issues/source?bid=2938665&fileBranchId=3246677#l86)